### PR TITLE
fix: Output width limit

### PR DESCRIPTION
**Pull request by Aleksander-Karlsson**
## Description
*Include a description of the change done*
- Ensure output data (not comments) are below character width limit of 132.
- Line break default multi-word header names. 
- Enable check on ~100 tests checking for width of output.

# Why
*The motivation for the change (if applicable)*
Some reservoir simulators using our output have character limits.

Close: #166 

# Checklist:
Before submitting this PR, please make sure:

- [x] I am complying with the [contributing](https://equinor.github.io/completor/contribution_guide) doc
- [x] My code builds clean without any errors or warnings
- [x] I have added/extended tests that prove my fix is effective or that my feature works
- [x] I have made corresponding changes to the documentation, and it builds correctly

### CODE REVIEW

File: completor/prepare_outputs.py [hunk 1]
```diff
@@ -91,42 +91,121 @@ def dataframe_tostring(
...
-        try:
-            output_string = df_temp.to_string(index=False, justify="justify", formatters=formatters, header=header)
-        except ValueError:
-            pass
+        formatters = {
+            Headers.STRENGTH: "{:.10g}".format,
+            Headers.SCALE_FACTOR: "{:.10g}".format,
+            Headers.ROUGHNESS: "{:.10g}".format,
+            Headers.CONNECTION_FACTOR: "{:.10g}".format,
+            "CONNECTION_FACTOR": "{:.10g}".format,
```
Comment (MegaAnda123) [inline completor/prepare_outputs.py:106]: Add header variations to Headers in constants.py and use them from there?
Maybe add a default format to the headers as well and get them from there?

File: completor/prepare_outputs.py [hunk 2]
```diff
@@ -91,42 +91,121 @@ def dataframe_tostring(
...
+            Headers.ALPHA_MAIN: "{:.10g}".format,
+            Headers.ALPHA_PILOT: "{:.10g}".format,
+        }
+
+        if header:
+            # Modify headers to reduce width.
+            column_splits = [tuple(column.split("_")) for column in df_temp.columns]
+            number_of_levels = max([len(tup) for tup in column_splits])
+            if number_of_levels > 1:
+                formatters.update(
```
Comment (MegaAnda123) [inline completor/prepare_outputs.py:139]: I get a linter warning that mapping does not have update as a method, I think dict concatenation/merging does the same thing e.g. formatters = {**formatters, **{<new dict>}}.

File: completor/prepare_outputs.py [hunk 3]
```diff
@@ -91,42 +91,121 @@ def dataframe_tostring(
...
+        # Probably find columns where this is the case and cast to numeric after replacing with nan?
+        df_temp[columns_with_1_star] = df_temp[columns_with_1_star].astype(np.float64, errors="ignore")
+        output_string = df_temp.to_string(
+            index=False, justify="justify", formatters=formatters, header=header, sparsify=False, na_rep="1*"
+        )
+
     if output_string is None:
         return ""
+
+    limit = 132
```
Comment (MegaAnda123) [inline completor/prepare_outputs.py:190]: This should be max 128, maybe even lower just to be safe?

**Review by MegaAnda123 — APPROVED** [review]
Some comments.

File: completor/prepare_outputs.py (hunk 3 above)
Comment (Aleksander-Karlsson) [reply completor/prepare_outputs.py]: Maxed it at 128!

File: completor/prepare_outputs.py (hunk 1 above)
Comment (Aleksander-Karlsson) [reply completor/prepare_outputs.py:104]: See: https://github.com/equinor/completor/issues/168

File: completor/prepare_outputs.py [hunk 4]
```diff
@@ -91,42 +89,120 @@ def dataframe_tostring(
...
+                df_temp.columns = pd.MultiIndex.from_frame(new_cols)
+
+    try:
+        output_string = df_temp.to_string(
+            index=False, justify="justify", formatters=formatters, header=header, sparsify=False
+        )
+    except ValueError:
+        if df_temp.isnull().values.any():
+            raise CompletorError("Got NaN values in table, please report if encountered!")
+        df_temp.replace("*", "1*", inplace=False)
```
Comment (Aleksander-Karlsson) [inline completor/prepare_outputs.py:176]:
```suggestion
        df_temp = df_temp.replace("*", "1*", inplace=False)
```

File: completor/utils.py [hunk 5]
```diff
@@ -231,3 +235,28 @@ def get_active_wells(completion_table: pd.DataFrame, gp_perf_devicelayer: bool)
...
+    Raises:
+        ValueError: If there exists any data that is too long.
+    """
+    lines = result.splitlines()
+    lengths = np.char.str_len(lines)
+    lines_to_check = np.nonzero(lengths >= limit)[0]
+    too_long_lines = []
+    for line_index in lines_to_check:
+        cleaned_line = lines[line_index].rsplit("/")[0] + "/"
+        cleaned_line = cleaned_line.rsplit("--")[0] + "--"
```
Comment (Aleksander-Karlsson) [inline completor/utils.py:258]:
```suggestion
        cleaned_line = cleaned_line.rsplit("--")[0]
```

Comment (Aleksander-Karlsson) [inline completor/utils.py:258]:
```suggestion
        cleaned_line = lines[line_index].rsplit("/", maxsplit=1)[0] + "/"
        cleaned_line = cleaned_line.rsplit("--", maxsplit=1)[0]
```

**Pull request merged**